ql/src/java/org/apache/hadoop/hive/ql/udf/generic/InstantDateTimeFormatter.java

```diff
@@ -49,7 +54,7 @@ public Instant parse(String text, String pattern) {
     Objects.requireNonNull(text);
     Objects.requireNonNull(pattern);
     Timestamp timestamp;
-    DateTimeFormatter formatter = getFormatter(pattern);
+    DateTimeFormatter formatter = getFormatter(pattern).withResolverStyle(ResolverStyle.STRICT);
```

Comment:
This is going to create a copy of the `DateTimeFormatter` for every call to `parse` beating the purpose of the cache. Why we can't set the style in the constructor?

Reply:
My bad, i have it fixed.